src/snowcli/cli/__init__.py

```diff
@@ -16,17 +16,30 @@
     streamlit,
 )
 from snowcli.cli.common.flags import DEFAULT_CONTEXT_SETTINGS
+from snowcli.cli.common.snow_cli_global_context import (
+    snow_cli_global_context_holder,
+    SnowCliGlobalContext,
+)
+from snowcli.cli.common.snow_cli_typer import SnowCliTyper
 from snowcli.cli.snowpark import app as snowpark_app
 from snowcli.config import AppConfig
 from snowcli.output.formats import OutputFormat
 from snowcli.connection_config import ConnectionConfigs
 
-app = typer.Typer(
+app = SnowCliTyper(
     context_settings=DEFAULT_CONTEXT_SETTINGS,
     pretty_exceptions_show_locals=False,
 )
 
 
+def setup_global_context(debug: bool):
```

Comment:
Can you please add short docstring about what this function does?

Reply:
Ok, I'll do it.

Reply:
As we agreed to not use the solution from this PR, I made changes from this comment in new PR: https://github.com/Snowflake-Labs/snowcli/pull/192